src/AccessibilityInsights.VersionSwitcher/MainWindow.xaml.cs

```diff
@@ -36,21 +36,37 @@ public MainWindow()
 
         public void SwitchVersionAndInvokeCloseApplication()
         {
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            ExecutionHistory history = new ExecutionHistory();
             string errorMessage = null;
 
             try
             {
-                InstallationEngine engine = new InstallationEngine(ProductName, SafelyGetAppInstalledPath());
-                engine.PerformInstallation(DispatcherUpdateProgress);
+                history.TypedExecutionResult = ResultExecutionWrapper.Execute(ExecutionResult.Unknown, 
+                    () => Properties.Resources.UnableToCompleteInstallation,
+                    () =>
+                    {
+                        InstallationEngine engine = new InstallationEngine(ProductName, SafelyGetAppInstalledPath(history), history);
+                        return engine.PerformInstallation(DispatcherUpdateProgress);
+                    });
             }
 #pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
We don't need this anymore, now that we're catching ResultsBearingExceptions

Reply:
Removed